studiopress/genesis-custom-blocks
# 'Edit Block' UI: Render block field

# Changes
* Uses Rob's markup and styling
* Opts out of the block editor for the 'Edit Block' UI
* Renders a React app in place of it
* Restructures the JS files into `src/` and `dist/`. Now, this will clear `dist/` after each build, so we don't have to worry about outdated files
* Targets `feature/react-ui`, which I just branched off `develop`. This will allow us to merge bugfixes into `develop` and release those.

# Testing instructions
1. `composer install && npm ci && npm run dev`
1. `wp post create --post_type='genesis_custom_block' --post_title='Test UI' --post_excerpt='' --post_content='{"genesis-custom-blocks\/test-ui":{"name":"test-ui","title":"Test UI","excluded":[],"icon":"genesis_custom_blocks","category":{"slug":"common","title":"Common Blocks","icon":null},"keywords":[""],"fields":{"text":{"name":"text","label":"Text","control":"text","type":"string","order":0,"location":"editor","width":"100","help":"","default":"","placeholder":"","maxlength":null}}}}'`
1. It should say something like `Success! Created post 8`
1. Go to the URL for that post: https://your-wp-local/wp-admin/post.php?post=8&action=edit
1. Expected: It should initially look like this:

![on-first-loading](https://user-images.githubusercontent.com/4063887/98227272-b881ed00-1f1c-11eb-9532-9808c6edb27e.png)

6. Change the 'Field' `<input>`:

![settings-in-ui](https://user-images.githubusercontent.com/4063887/98228055-b3716d80-1f1d-11eb-9cbf-4fbcb7f26681.gif)

7. Expected: the settings should change, depending on what field you changed to
8. The editor has a lot of work needed. This PR is mainly for the field settings, and it only handles 1 field, not multiple, as there's no way to add more fields yet.

![working-here-now](https://user-images.githubusercontent.com/4063887/98228238-f0d5fb00-1f1d-11eb-8ba6-b94bf3528641.png)
